c7n/resources/ec2.py

```diff
@@ -206,6 +206,34 @@ def get_resource_date(self, i):
         return None
 
 
+class ASGFilter(object):
+    """Filter instances by whether they are members of an ASG"""
+    asg_member = None
+
+    def filter_asg_membership(self, instances, desired=None):
+        if desired is None:
+            desired = self.asg_member
+
+        if self.asg_member is None:
+            # TODO raise exception
```

Comment:
Is there a case where someone would want an `autorecover-alarm` on an ASG instance? I didn't think so, which is why I wanted to reuse the logic to exclude ASG instances in that action that I had used in the `singleton` filter. The `ASGFilter` class is there only to reuse the logic. I'm open to other ideas. Please advise.